upf/transformers/negative_conditions_remover.py

```diff
@@ -67,8 +72,10 @@ def get_rewritten_problem(self) -> Problem:
         if self._new_problem is not None:
             return self._new_problem
         #NOTE that a different environment might be needed when multi-threading
-        self._create_problem_copy('no_negative_conditions')
-        self._new_problem_add_objects()
+
+        self._new_problem = Problem(f'{self._name}_{self._problem.name}', self._env)
+        for o in self._problem.all_objects():
+            self._new_problem.add_object(o)
         assert self._new_problem is not None
         self._modify_actions_and_goals()
```

Comment:
In this method you are doing all the transformation

Reply:
What do you mean?

Reply:
All the transformation code is inside the `_modify_actions_and_goals` method.
You are not only modifying actions and goals.

Reply:
Oh ok, so you mean I should update the name of the method to something more appropriate!

Reply:
No, I think that if possible you should refactor the code in more methods.
Otherwise the code can simply stay in the `get_rewritten_problem` method.

Reply:
Yes, makes sense